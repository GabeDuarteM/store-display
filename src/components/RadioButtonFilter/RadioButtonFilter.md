```js
initialState = {
  options: [{ id: 'woman', text: 'Woman' }, { id: 'man', text: 'Man' }],
  activeOption: 'woman',
}
;<RadioButtonFilter
  onChange={id => {
    setState({
      activeOption: id,
    })
  }}
  options={state.options}
  activeOption={state.activeOption}
/>
```
