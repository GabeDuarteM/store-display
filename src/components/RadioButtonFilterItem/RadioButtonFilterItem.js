import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import cn from 'classnames'

export const styles = {
  radioButtonFilterItem: {
    boxSizing: 'border-box',
    fontSize: 12,
    width: 160,
    height: 36,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    color: '#656565',
    backgroundColor: 'white',
    border: '1px solid #d7d8db',
    borderLeft: '0',
    borderRight: '0',
    '&:first-child': {
      borderLeft: '1px solid #d7d8db',
      borderTopLeftRadius: '3px',
      borderBottomLeftRadius: '3px',
    },
    '&:last-child': {
      borderRight: '1px solid #d7d8db',
      borderTopRightRadius: '3px',
      borderBottomRightRadius: '3px',
    },
    '&.active': {
      color: 'white',
      backgroundColor: '#6F8CE3',
      border: '1px solid #6F8CE3 !important',
    },
  },
  hiddenInput: {
    display: 'none',
  },
}

const RadioButtonFilterItem = ({ classes, text, active, onChange }) => (
  <label className={cn(classes.radioButtonFilterItem, { active })}>
    <input
      className={classes.hiddenInput}
      type="radio"
      checked={active}
      onChange={onChange}
    />
    {text}
  </label>
)

export const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
  /** Text to be displayed */
  text: PropTypes.string.isRequired,
  /** Property to define if the filter is active or not */
  active: PropTypes.bool,
  /**
   * Gets called when the user clicks on the radiobutton
   *
   * @param {string} id Id of the option that was clicked
   */
  onChange: PropTypes.func.isRequired,
}

export const defaultProps = {
  active: false,
}

RadioButtonFilterItem.propTypes = propTypes

RadioButtonFilterItem.defaultProps = defaultProps

export default injectSheet(styles)(RadioButtonFilterItem)
