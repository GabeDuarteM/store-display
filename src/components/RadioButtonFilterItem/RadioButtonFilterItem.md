Active

```js
<RadioButtonFilterItem text="Text" onChange={() => {}} active />
```

Inactive

```js
<RadioButtonFilterItem text="Text" onChange={() => {}} />
```
