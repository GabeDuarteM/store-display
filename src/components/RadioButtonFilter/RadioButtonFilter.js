import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import RadioButtonFilterItem from '../RadioButtonFilterItem'

export const styles = {
  radioButtonFilter: {
    display: 'flex',
    userSelect: 'none',
  },
}

const RadioButtonFilter = ({ classes, options, activeOption, onChange }) => (
  <div className={classes.radioButtonFilter}>
    {options.map(elem => (
      <RadioButtonFilterItem
        key={elem.id}
        onChange={() => onChange(elem.id)}
        text={elem.text}
        active={elem.id === activeOption}
      />
    ))}
  </div>
)

export const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
  /** Array of items that will be displayed inside the RadioButton */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
    }),
  ).isRequired,
  /**
   * Gets called when the user clicks on the radiobutton
   *
   * @param {string} id Id of the option that was clicked
   */
  onChange: PropTypes.func.isRequired,
  /** The active filter */
  activeOption: PropTypes.string,
}

export const defaultProps = {
  activeOption: null,
}

RadioButtonFilter.propTypes = propTypes
RadioButtonFilter.defaultProps = defaultProps

export default injectSheet(styles)(RadioButtonFilter)
