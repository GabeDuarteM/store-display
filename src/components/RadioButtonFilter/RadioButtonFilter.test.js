import React from 'react'

import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import RadioButtonFilter, { styles, propTypes } from './RadioButtonFilter'
import RadioButtonFilterItem from '../RadioButtonFilterItem'

const createWrapper = (func, optionalProps) =>
  func(
    <RadioButtonFilter
      onChange={jest.fn}
      options={[{ id: 'woman', text: 'Woman' }, { id: 'man', text: 'Man' }]}
      activeOption="woman"
      {...optionalProps}
    />,
  )

const createWrapperShallow = optionalProps =>
  createWrapper(shallow, optionalProps)

const createWrapperMount = optionalProps => createWrapper(mount, optionalProps)

describe('COMPONENT: <RadioButtonFilter />', () => {
  it('should render without crashing with default props', () => {
    createWrapperMount()
  })

  it("should match component's snapshot", () => {
    const wrapper = createWrapperShallow()
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it("should match component's style snapshot", () => {
    expect(styles).toMatchSnapshot()
  })

  it("should match component's propTypes snapshot", () => {
    expect(propTypes).toMatchSnapshot()
  })

  it('should render one RadioButtonFilterItem component for each option passed', () => {
    const options = [
      { id: 'woman', text: 'Woman' },
      { id: 'man', text: 'Man' },
      { id: 'alien', text: 'Alien' },
    ]
    const wrapper = createWrapperShallow({ options })

    expect(wrapper.dive().find(RadioButtonFilterItem).length).toBe(3)
  })

  it('should render RadioButtonFilterItem with the props passed on the options', () => {
    expect.assertions(9)
    const options = [
      { id: 'woman', text: 'Woman' },
      { id: 'man', text: 'Man' },
      { id: 'alien', text: 'Alien' },
    ]
    const activeOption = 'man'
    const onChange = jest.fn()
    const wrapper = createWrapperShallow({ options, activeOption, onChange })
    const elements = wrapper.dive().find(RadioButtonFilterItem)

    for (let i = 0; i < options.length; i += 1) {
      expect(elements.at(i).prop('text')).toBe(options[i].text)
      expect(elements.at(i).prop('active')).toBe(options[i].id === activeOption)
      const radioButtonFilterItemOnChange = elements.at(i).prop('onChange')
      radioButtonFilterItemOnChange()
      expect(onChange).toHaveBeenLastCalledWith(options[i].id)
    }
  })
})
