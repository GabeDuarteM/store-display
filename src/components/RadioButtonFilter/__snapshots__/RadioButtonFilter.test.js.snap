// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: <RadioButtonFilter /> should match component's propTypes snapshot 1`] = `
Object {
  "activeOption": [Function],
  "classes": [Function],
  "onChange": [Function],
  "options": [Function],
}
`;

exports[`COMPONENT: <RadioButtonFilter /> should match component's snapshot 1`] = `
<div
  className="RadioButtonFilter-radioButtonFilter"
>
  <Jss(RadioButtonFilterItem)
    active={true}
    key="woman"
    onChange={[Function]}
    text="Woman"
  />
  <Jss(RadioButtonFilterItem)
    active={false}
    key="man"
    onChange={[Function]}
    text="Man"
  />
</div>
`;

exports[`COMPONENT: <RadioButtonFilter /> should match component's style snapshot 1`] = `
Object {
  "radioButtonFilter": Object {
    "display": "flex",
    "userSelect": "none",
  },
}
`;
